BUG: .flags in nddata.CCDData.to_hdu dtype check fails in 32-bit (armv7)

Caused by https://github.com/astropy/astropy/pull/18862 (cc @BrunoSanchez)

Example log: https://github.com/astropy/astropy/actions/runs/20423538327/job/58679497060

```
 _____________________ test_write_read_with_flag_collection _____________________
  
  tmp_path = PosixPath('/tmp/pytest-of-root/pytest-0/test_write_read_with_flag_coll0')
  
      def test_write_read_with_flag_collection(tmp_path):
          # Test that if flags are present as a FlagCollection they
          # are saved and loaded by default.
          ccd_data = create_ccd_data()
      
          # Create a FlagCollection with different flag types
          flags = FlagCollection(shape=ccd_data.data.shape)
          # Add different types of flags
          flags["BAD_PIXEL"] = np.zeros_like(ccd_data.data, dtype=bool)
          flags["SATURATED"] = np.zeros_like(ccd_data.data, dtype=int)
          flags["BAD_PIXEL"][50:60, 50:60] = True  # Mark a region as bad
      
          ccd_data.flags = flags
      
          filename = str(tmp_path / "a_file.fits")
          ccd_data.write(filename, hdu_flags="FLAGS")
          ccd_after = CCDData.read(filename, hdu_flags="FLAGS")
      
          assert ccd_after.flags is not None
          assert isinstance(ccd_after.flags, FlagCollection)
          assert len(ccd_after.flags.keys()) == len(ccd_data.flags.keys())
          assert ccd_after.flags.shape == ccd_data.flags.shape
          assert ccd_after.flags["BAD_PIXEL"].dtype == np.uint8
  >       assert ccd_after.flags["SATURATED"].dtype == ">i8"
  E       AssertionError: assert dtype('>i4') == '>i8'
  E        +  where dtype('>i4') = array(..., dtype='>i4').dtype
  astropy/nddata/tests/test_ccddata.py:1199: AssertionError
```

--- a/astropy/nddata/tests/test_ccddata.py
+++ b/astropy/nddata/tests/test_ccddata.py
@@ -1,6 +1,7 @@
 # Licensed under a 3-clause BSD style license - see LICENSE.rst
 
 import os
+import sys
 import textwrap
 
 import numpy as np
@@ -1196,7 +1197,12 @@ def test_write_read_with_flag_collection(tmp_path):
     assert len(ccd_after.flags.keys()) == len(ccd_data.flags.keys())
     assert ccd_after.flags.shape == ccd_data.flags.shape
     assert ccd_after.flags["BAD_PIXEL"].dtype == np.uint8
-    assert ccd_after.flags["SATURATED"].dtype == ">i8"
+    if sys.maxsize > 2**32:
+        expected_int_size = 8
+    else:
+        # 32bit arch
+        expected_int_size = 4
+    assert ccd_after.flags["SATURATED"].dtype == f">i{expected_int_size}"
     np.testing.assert_array_equal(
         ccd_data.flags["BAD_PIXEL"], ccd_after.flags["BAD_PIXEL"]
     )

